api/api.go

```diff
@@ -39,24 +39,28 @@ func Setup(router *mux.Router,
 	producer ReindexRequestedProducer) *API {
 	api := &API{
 		Router:      router,
+		cfg:         cfg,
 		dataStore:   dataStore,
 		permissions: permissions,
 		taskNames:   taskNames,
-		cfg:         cfg,
 		httpClient:  httpClient,
 		reindex:     reindex,
 		producer:    producer,
 	}
 
-	router.HandleFunc("/jobs", api.GetJobsHandler).Methods("GET")
-	router.HandleFunc("/jobs", api.CreateJobHandler).Methods("POST")
-	router.HandleFunc("/jobs/{id}", api.GetJobHandler).Methods("GET")
-	router.HandleFunc("/jobs/{id}", permissions.Require(update, api.PatchJobStatusHandler)).Methods("PATCH")
-	router.HandleFunc("/jobs/{id}/number_of_tasks/{count}", api.PutNumTasksHandler).Methods("PUT")
-	router.HandleFunc("/jobs/{id}/tasks", api.GetTasksHandler).Methods("GET")
-	taskHandler := permissions.Require(update, api.CreateTaskHandler)
-	router.HandleFunc("/jobs/{id}/tasks", taskHandler).Methods("POST")
-	router.HandleFunc("/jobs/{id}/tasks/{task_name}", api.GetTaskHandler).Methods("GET")
+	// These routes should always use the latest API version
+	api.latestAPIRoutes(permissions)
+
+	v1 := router.PathPrefix("/{version:v1}").Subrouter()
+	v1.HandleFunc("/jobs", api.GetJobsHandler).Methods("GET")
```

Comment:
You are duplicating code here which is unnecessary
Note that both the type of `v1` and `api.Router` are `*mux.Router`
So you can create a generic function which passes in a `*mux.Router` value (let's say `router`) and this function can then include 
`router.HandleFunc().....`

Then you can call this function twice where you pass in `api.Router` and then `v1` for the second